src/Function.cpp

```diff
@@ -441,6 +474,16 @@ Function::make_random_signature(const CGContext& cg_context, const Type* type, c
 
 	if(CGOptions::func_attr_aligned() && rnd_flipcoin(FuncAttrAligned))
 		f->func_attr_aligned = true;
+
+	if(CGOptions::func_attr_flag()){
```

Comment:
Please pass the values of `CGOptions` into derived `FunctionAttribute` classes and then you don't need to care of it in `make_random_signature`.

Reply:
And you can then remove `func_attr_flag`.

Reply:
`CGOptions` can be accessed throughout the program. It is required in `make_random_signature` because when command line is not set, it should not generate attributes unnecessary.

`func_attr_flag` is required in output phase to output attributes only if command line is set. However, `CGOption` can be used here, but in remaining extensions bool is used.

Reply:
Ideally you can add `Attributes` instances to `attributes` only when the corresponding attributes are enabled.

Reply:
Does it mean setting two different command lines for - 
1. `common_func_attributes` - bool attributes (output attributes)
2. multivalued_func_attributes - visibility, used/unused, no_sanitize

Reply:
I would do for now just a single and we'll see what will community say us.

Reply:
Got it. If instances are not pushed, there is nothing to print. Hence, it is fine to remove `func_attr_flag`